src/main/webapp/js/go-to-deal.js

```diff
@@ -113,7 +84,72 @@ function createCommentElement(commentEntity) {
   return commentElement;
 }
 
+let myVote = 0;
+/**
+ * Updates vote UI based on global variable myVote
+ */
+function updateMyVote() {
+  const upvoteBtn = document.getElementById('upvote-btn');
+  const downvoteBtn = document.getElementById('downvote-btn');
+  upvoteBtn.classList.remove('active');
+  downvoteBtn.classList.remove('active');
+  if (myVote > 0) {
+    upvoteBtn.classList.add('active');
+  } else if (myVote < 0) {
+    downvoteBtn.classList.add('active');
+  }
+  const voteElement = document.getElementById('votes-num');
+  voteElement.innerText = votes + myVote;
+}
+
+/**
+ * Called when the user clicks the upvote button
+ */
+function handleUpvote() {
+  if (myVote == 1) {
+    myVote = 0;
+  } else {
+    myVote = 1;
+  }
+  updateMyVote();
+}
+
+/**
+ * Called when the user clicks the downvote button
+ */
+function handleDownvote() {
+  if (myVote == -1) {
+    myVote = 0;
+  } else {
+    myVote = -1;
+  }
+  updateMyVote();
+}
+
+/**
+ * Display Deal Not Found on the page
+ */
+function showNotFound() {
+  $('#deal-loading').hide();
+  $('#deal-notfound').show();
+}
+
+/**
+ * Calls backend for data on deal
+ */
+function initDeal() {
+  const myPath = window.location.pathname; // path is /deals/<id>
+  const myId = myPath.substr(7);
+  $.ajax('/api/deals/' + myId)
+      .done((deal) => {
+        loadDealDataToPage(deal);
+      })
+      .fail(() => {
+        showNotFound();
+      });
+}
+
 addLoadEvent(() => {
-  getDeal(deal);
+  initDeal();
```

Comment:
Use the common function from `util.js` here?

Reply:
what common function? `addLoadEvent` is already from util